java-generator/core/src/test/java/io/fabric8/java/generator/GeneratorTest.java

```diff
@@ -390,6 +398,51 @@ void testObjectWithPreservedFields() {
     assertTrue(clzT.get().getFieldByName("additionalProperties").isPresent());
   }
 
+  @Test
+  void testObjectNullableFieldsManagement() {
+    // Arrange
+    Map<String, JSONSchemaProps> props = new HashMap<>();
+    JSONSchemaProps nullableObj = new JSONSchemaProps();
+    nullableObj.setType("object");
+    nullableObj.setNullable(Boolean.TRUE);
+    props.put("o1", nullableObj);
+
+    JSONSchemaProps nonNullableObj = new JSONSchemaProps();
+    nonNullableObj.setType("object");
+    nonNullableObj.setNullable(Boolean.FALSE);
+    props.put("o2", nonNullableObj);
+
+    JObject obj = new JObject(null, "t", props, null, false, "", "", defaultConfig, null, Boolean.FALSE);
+
+    // Act
+    GeneratorResult res = obj.generateJava();
+
+    // Assert
+    assertEquals(3, res.getTopLevelClasses().size());
+    assertEquals("O1", res.getTopLevelClasses().get(0).getName());
+    assertEquals("O2", res.getTopLevelClasses().get(1).getName());
+    assertEquals("T", res.getTopLevelClasses().get(2).getName());
+
+    Optional<ClassOrInterfaceDeclaration> clzT = res.getTopLevelClasses().get(2).getCompilationUnit().getClassByName("T");
+    assertTrue(clzT.isPresent());
+    assertEquals(2, clzT.get().getFields().size());
+    Optional<FieldDeclaration> o1Field = clzT.get().getFieldByName("o1");
+    assertTrue(o1Field.isPresent());
+    FieldDeclaration actualO1Field = o1Field.get();
+    Optional<AnnotationExpr> nullableAnnotation = actualO1Field
+        .getAnnotationByName("com.fasterxml.jackson.annotation.JsonSetter");
+    assertFalse(nullableAnnotation.isPresent());
+
+    Optional<FieldDeclaration> o2Field = clzT.get().getFieldByName("o2");
+    assertTrue(o2Field.isPresent());
+    FieldDeclaration actualO2Field = o2Field.get();
+    nullableAnnotation = actualO2Field.getAnnotationByName("com.fasterxml.jackson.annotation.JsonSetter");
+    assertTrue(nullableAnnotation.isPresent());
```

Comment:
can we check the actual value of nulls of JsonSetter too?

Reply:
+1, done.